src/components/EmojiPicker/EmojiPickerMenu/index.tsx

```diff
@@ -139,14 +140,6 @@ function EmojiPickerMenu({onEmojiSelected, activeEmoji}: EmojiPickerMenuProps, r
                 return;
             }
 
-            if (!isEnterWhileComposition(keyBoardEvent) && keyBoardEvent.key === CONST.KEYBOARD_SHORTCUTS.ENTER.shortcutKey) {
-                // On web, avoid this Enter default input action; otherwise, it will add a new line in the subsequently focused composer.
-                keyBoardEvent.preventDefault();
-                // On mWeb, avoid propagating this Enter keystroke to Pressable child component; otherwise, it will trigger the onEmojiSelected callback again.
-                keyBoardEvent.stopPropagation();
-                return;
-            }
```

Comment:
Why you remove the whole condition, Shouldn't we keep the return early here?

Reply:
I removed it because it won't affect the code below it.
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L150-L161

For
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L157-L160

it's only executed if `shouldAutoFocusOnKeyPress` is true which is false for ENTER.
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/libs/ReportUtils.ts#L7595-L7598

For
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L150-L155

the keyboard movement state is used to show the focused state of emoji item.
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L271

The emoji will be focused when we use arrow key, tab, or hold down mouse on it. The time we press Enter, the emoji will be selected. From my understanding, the `isUsingKeyboardMovement` state is used to not show the blue outline on the emoji when focused using mouse.

But it will still show anyway with the mouse because when focused, we update the focused index.
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L293

which always update `isUsingKeyboardMovement` to true.
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L75-L87

We can remove Enter from the `if` if you want.
https://github.com/Expensify/App/blob/c3d33fcec7df3fcbb31cdfcb61951ce985da63c7/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L153

Reply:
> We can remove Enter from the `if` if you want.

To be honest, I don't understand what is the purpose of `keyBoardEvent.key === 'Enter' ||`, As I see in the code clicking `Enter` should do one of 
1. select the emoji and close the menu
2. select the category

In the both cases this condition `keyBoardEvent.key === 'Enter' ||` it is not reachable in the old code [here](https://github.com/Expensify/App/pull/47761/files#diff-fdc9a419f8f1bb009124ad48fc21a1a09bded636e1c3533fa467ad02ef463e97R150), and beside your explanation about `isUsingKeyboardMovement` it seems to don't do any function but let's know what @aldo-expensify think before remove it

Reply:
> In the both cases this condition keyBoardEvent.key === 'Enter' || it is not reachable in the old code [here](https://github.com/Expensify/App/pull/47761/files#diff-fdc9a419f8f1bb009124ad48fc21a1a09bded636e1c3533fa467ad02ef463e97R150)

Yeah, good point.

Reply:
Sorry for the delay here, I'm just starting to look at this 👁️

Reply:
I'm very unfamiliar with the details here, and I'm wanting to avoid spending to much time to understand all the details around it. I tested and it appears to be working well for me, so I'm going to just assume that we don't need to change anything here since it is working.